### PR TITLE
YD-203 Added activity overview self-links

### PR DESCRIPTION
**Pull request by Bert-R**
For:
* User activity overviews (both day and week), with and without buddies
* Buddy activity overviews (both day and week)

Extended the integration tests accordingly

### CODE REVIEW

File: appservice/src/main/java/nu/yona/server/analysis/rest/ActivityControllerBase.java [hunk 1]
```diff
@@ -79,12 +81,14 @@
 	protected static final int WEEKS_DEFAULT_PAGE_SIZE = 2;
 	protected static final int DAYS_DEFAULT_PAGE_SIZE = 3;
 	protected static final int MESSAGES_DEFAULT_PAGE_SIZE = 4;
-	protected static final String PREV_REL = "prev"; // IANA reserved, so will not be prefixed
```
Comment (aristotelos) [inline appservice/src/main/java/nu/yona/server/analysis/rest/ActivityControllerBase.java]: There are many whitespace changes in this PR. Any idea why?

File: appservice/src/main/resources/static/swagger/swagger-spec.yaml [hunk 2]
```diff
@@ -603,6 +648,21 @@ paths:
           description: Activity of the buddy.
           schema:
             $ref: '#/definitions/PagedDayActivityOverviews'
+  '/users/{userID}/buddies/{buddyID}/activity/days/{date}':
+    get:
+      summary: Get a day activity overview
+      description: Get network and app activity of the specified day
+      parameters:
+        - $ref: "#/parameters/yonaPassword"
```
Comment (aristotelos) [inline appservice/src/main/resources/static/swagger/swagger-spec.yaml]: `buddyID` is missing

File: appservice/src/main/resources/static/swagger/swagger-spec.yaml [hunk 3]
```diff
@@ -819,6 +879,21 @@ paths:
           description: Activity of the buddy.
           schema:
             $ref: '#/definitions/PagedWeekActivityOverviews'
+  '/users/{userID}/buddies/{buddyID}/activity/weeks/{week}':
+    get:
+      summary: Get a week activity overview
+      description: Get network and app activity of the specified week
+      parameters:
+        - $ref: "#/parameters/yonaPassword"
```
Comment (aristotelos) [inline appservice/src/main/resources/static/swagger/swagger-spec.yaml]: `buddyID` is missing

File: appservice/src/main/resources/static/swagger/swagger-spec.yaml (hunk 2 above)
Comment (Bert-R) [inline appservice/src/main/resources/static/swagger/swagger-spec.yaml]: Sharp! I'll add it.

File: appservice/src/main/resources/static/swagger/swagger-spec.yaml (hunk 3 above)
Comment (Bert-R) [inline appservice/src/main/resources/static/swagger/swagger-spec.yaml]: This one too.

File: appservice/src/main/java/nu/yona/server/analysis/rest/ActivityControllerBase.java (hunk 1 above)
Comment (Bert-R) [inline appservice/src/main/java/nu/yona/server/analysis/rest/ActivityControllerBase.java]: I don't think there are many. This comment was wrapped in an earlier formatting action (with a different formatter) and not joined again when reformatted with the Yona formatter. Corrected that and also the next one.

Besides this, I see a line wrapping change, but that was originally too long and now corrected.

**Comment by Bert-R** [comment]
Pushed an update.

**Pull request merged**